presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/Rule.java

```diff
@@ -46,4 +48,40 @@ default boolean isEnabled(Session session)
 
         Session getSession();
     }
+
+    class Result
+    {
+        @Deprecated
+        public static Result fromOptional(Optional<PlanNode> transformedPlan)
+        {
+            return new Result(transformedPlan);
+        }
+
+        public static Result empty()
+        {
+            return new Result(Optional.empty());
+        }
+
+        public static Result replace(PlanNode transformedPlan)
```

Comment:
I think it should be called `addPlanNode`, because other kind of calls are rather optimizations (per @martint  discussions) and can be part of rule manifest. Maybe `of` would be fine too, but I think it's too anonymous.

`addPlanNode` is the most basic operation for a rule

Reply:
+1